stdlib/array.ml

```diff
@@ -441,12 +441,10 @@ let fast_sort = stable_sort
 
 let shuffle_contract_violation i j =
   let int = string_of_int in
-  String.concat "" [
-    "Array.shuffle: 'rand "; int (i + 1);
-    "' returned "; int j;
-    ", out of expected range [0; "; int i; "]"
-  ]
-  |> invalid_arg
+  invalid_arg
+    ("Array.shuffle: 'rand " ^ int (i + 1) ^
+     "' returned " ^ int j ^
+     ", out of expected range [0; " ^ int i ^ "]")
```

Comment:
This change seems unnecessary. But OK to leave it in if you feel strongly about it :)

Reply:
Not sure I understand, did I change something about the error message ? Otherwise I need this change: `Array` was depending on `String` only because of the `String.concat` there, with this PR `String` depends on `Array` so I need to break the loop.

Reply:
I had missed the technical point about breaking the dependency loop. Sorry for the noise.

Reply:
(A short comment `(* not using String.concat to avoid an Array->String dependency *)` would possibly have been helpful here.)